### PR TITLE
Add file ref support

### PR DESCRIPTION
**Pull request by Deadarius**
- Refactor: replace supplied utils with lodash
- Type improvements
- Use `json-schema-ref-parser` to combine bundle schemas while preserving $refs 
- Now schema can be supplied as a path to file

Lint errors to be fixed

### CODE REVIEW

File: src/DefinitionGenerator.ts [hunk 1]
```diff
@@ -1,10 +1,11 @@
-import { JSONSchema7 } from 'json-schema';
+import _ = require('lodash');
```
Comment (lwcooper) [inline src/DefinitionGenerator.ts:1]: this syntax looks weird. how does it work? I've never seen it before

also, we normally use ramda 🙈 But I can see how much it makes sense to just replace the existing utils with lodash here 👍

File: src/ServerlessOpenApiDocumentation.ts [hunk 2]
```diff
@@ -1,31 +1,58 @@
...
-  private customVars;
+  private customVars: ICustomVars;
 
   /**
    * Constructor
    * @param serverless
    * @param options
    */
-  constructor (serverless, options) {
+  constructor (serverless: IFullServerless, options) {
```
Comment (lwcooper) [inline src/ServerlessOpenApiDocumentation.ts:52]: how come `options` doesn't need a type?

**Review by lwcooper — APPROVED** [review]
Lack of tests is a bit worrying in this whole module really! But not gonna block this PR being merged

File: src/ServerlessOpenApiDocumentation.ts (hunk 2 above)
Comment (Deadarius) [reply src/ServerlessOpenApiDocumentation.ts:52]: This project is not configured(yet?) to use `strict` typescript transpilation.

File: src/DefinitionGenerator.ts (hunk 1 above)
Comment (Deadarius) [reply src/DefinitionGenerator.ts:1]: This is an open source project, I would prefer to keep it easy for people to contribute - `lodash` is more aligned with this idea than `ramda`

**Comment by Deadarius** [comment]
RE: test covereage - https://github.com/conqa/serverless-openapi-documentation/issues/13

**Pull request merged**